{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "138d1e7c-cb1f-435a-b59f-37bd3a07709f",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "161c80fc-4b8b-433f-b24b-0967702e508a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>molecular_profile</th>\n",
       "      <th>molecular_profile_id</th>\n",
       "      <th>disease</th>\n",
       "      <th>doid</th>\n",
       "      <th>phenotypes</th>\n",
       "      <th>therapies</th>\n",
       "      <th>therapy_interaction_type</th>\n",
       "      <th>evidence_type</th>\n",
       "      <th>evidence_direction</th>\n",
       "      <th>evidence_level</th>\n",
       "      <th>...</th>\n",
       "      <th>citation</th>\n",
       "      <th>nct_ids</th>\n",
       "      <th>rating</th>\n",
       "      <th>evidence_status</th>\n",
       "      <th>evidence_id</th>\n",
       "      <th>variant_origin</th>\n",
       "      <th>last_review_date</th>\n",
       "      <th>evidence_civic_url</th>\n",
       "      <th>molecular_profile_civic_url</th>\n",
       "      <th>is_flagged</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>JAK2 V617F</td>\n",
       "      <td>64</td>\n",
       "      <td>Lymphoid Leukemia</td>\n",
       "      <td>1037</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Supports</td>\n",
       "      <td>B</td>\n",
       "      <td>...</td>\n",
       "      <td>Levine et al., 2005</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4.0</td>\n",
       "      <td>accepted</td>\n",
       "      <td>1</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2023-01-09 21:46:26 UTC</td>\n",
       "      <td>https://civicdb.org/links/evidence_items/1</td>\n",
       "      <td>https://civicdb.org/links/molecular_profiles/64</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PDGFRA D842V</td>\n",
       "      <td>99</td>\n",
       "      <td>Gastrointestinal Stromal Tumor</td>\n",
       "      <td>9253</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Supports</td>\n",
       "      <td>B</td>\n",
       "      <td>...</td>\n",
       "      <td>Lasota et al., 2004</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>accepted</td>\n",
       "      <td>2</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2023-01-09 21:46:27 UTC</td>\n",
       "      <td>https://civicdb.org/links/evidence_items/2</td>\n",
       "      <td>https://civicdb.org/links/molecular_profiles/99</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DNMT3A R882</td>\n",
       "      <td>32</td>\n",
       "      <td>Acute Myeloid Leukemia</td>\n",
       "      <td>9119</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Supports</td>\n",
       "      <td>B</td>\n",
       "      <td>...</td>\n",
       "      <td>LaRochelle et al., 2011</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.0</td>\n",
       "      <td>accepted</td>\n",
       "      <td>3</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2023-01-09 21:46:25 UTC</td>\n",
       "      <td>https://civicdb.org/links/evidence_items/3</td>\n",
       "      <td>https://civicdb.org/links/molecular_profiles/32</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DNMT3A R882</td>\n",
       "      <td>32</td>\n",
       "      <td>Acute Myeloid Leukemia</td>\n",
       "      <td>9119</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Supports</td>\n",
       "      <td>B</td>\n",
       "      <td>...</td>\n",
       "      <td>Ribeiro et al., 2012</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>accepted</td>\n",
       "      <td>4</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2023-01-09 21:46:25 UTC</td>\n",
       "      <td>https://civicdb.org/links/evidence_items/4</td>\n",
       "      <td>https://civicdb.org/links/molecular_profiles/32</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4 rows Ã— 25 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  molecular_profile  molecular_profile_id                         disease  \\\n",
       "0        JAK2 V617F                    64               Lymphoid Leukemia   \n",
       "1      PDGFRA D842V                    99  Gastrointestinal Stromal Tumor   \n",
       "2       DNMT3A R882                    32          Acute Myeloid Leukemia   \n",
       "3       DNMT3A R882                    32          Acute Myeloid Leukemia   \n",
       "\n",
       "   doid phenotypes therapies therapy_interaction_type evidence_type  \\\n",
       "0  1037        NaN       NaN                      NaN    Diagnostic   \n",
       "1  9253        NaN       NaN                      NaN    Diagnostic   \n",
       "2  9119        NaN       NaN                      NaN    Diagnostic   \n",
       "3  9119        NaN       NaN                      NaN    Diagnostic   \n",
       "\n",
       "  evidence_direction evidence_level  ...                 citation nct_ids  \\\n",
       "0           Supports              B  ...      Levine et al., 2005     NaN   \n",
       "1           Supports              B  ...      Lasota et al., 2004     NaN   \n",
       "2           Supports              B  ...  LaRochelle et al., 2011     NaN   \n",
       "3           Supports              B  ...     Ribeiro et al., 2012     NaN   \n",
       "\n",
       "  rating evidence_status  evidence_id variant_origin         last_review_date  \\\n",
       "0    4.0        accepted            1        Somatic  2023-01-09 21:46:26 UTC   \n",
       "1    3.0        accepted            2        Somatic  2023-01-09 21:46:27 UTC   \n",
       "2    2.0        accepted            3        Somatic  2023-01-09 21:46:25 UTC   \n",
       "3    3.0        accepted            4        Somatic  2023-01-09 21:46:25 UTC   \n",
       "\n",
       "                           evidence_civic_url  \\\n",
       "0  https://civicdb.org/links/evidence_items/1   \n",
       "1  https://civicdb.org/links/evidence_items/2   \n",
       "2  https://civicdb.org/links/evidence_items/3   \n",
       "3  https://civicdb.org/links/evidence_items/4   \n",
       "\n",
       "                       molecular_profile_civic_url  is_flagged  \n",
       "0  https://civicdb.org/links/molecular_profiles/64       False  \n",
       "1  https://civicdb.org/links/molecular_profiles/99       False  \n",
       "2  https://civicdb.org/links/molecular_profiles/32       False  \n",
       "3  https://civicdb.org/links/molecular_profiles/32       False  \n",
       "\n",
       "[4 rows x 25 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#load civic data\n",
    "raw_df = pd.read_csv(\"civic_raw.tsv\", sep='\\t')\n",
    "raw_df['doid'] = raw_df['doid'].fillna(0).astype('int64')\n",
    "raw_df.head(4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "d9eb19b4-950a-41d3-bb09-14e42d9ef65a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['molecular_profile', 'molecular_profile_id', 'disease', 'doid',\n",
       "       'phenotypes', 'therapies', 'therapy_interaction_type', 'evidence_type',\n",
       "       'evidence_direction', 'evidence_level', 'significance',\n",
       "       'evidence_statement', 'citation_id', 'source_type', 'asco_abstract_id',\n",
       "       'citation', 'nct_ids', 'rating', 'evidence_status', 'evidence_id',\n",
       "       'variant_origin', 'last_review_date', 'evidence_civic_url',\n",
       "       'molecular_profile_civic_url', 'is_flagged'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#check column names and delete unnecessary columns\n",
    "raw_df.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "8d516795-fb41-4d4b-a283-d36de6321993",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>molecular_profile</th>\n",
       "      <th>molecular_profile_id</th>\n",
       "      <th>disease</th>\n",
       "      <th>doid</th>\n",
       "      <th>evidence_type</th>\n",
       "      <th>evidence_statement</th>\n",
       "      <th>citation_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>JAK2 V617F</td>\n",
       "      <td>64</td>\n",
       "      <td>Lymphoid Leukemia</td>\n",
       "      <td>1037</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>JAK2 V617F is not associated with lymphoid leu...</td>\n",
       "      <td>16081687</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PDGFRA D842V</td>\n",
       "      <td>99</td>\n",
       "      <td>Gastrointestinal Stromal Tumor</td>\n",
       "      <td>9253</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>GIST tumors harboring PDGFRA D842V mutation ar...</td>\n",
       "      <td>15146165</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DNMT3A R882</td>\n",
       "      <td>32</td>\n",
       "      <td>Acute Myeloid Leukemia</td>\n",
       "      <td>9119</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>DNMT3A R882 mutations occur most often in de n...</td>\n",
       "      <td>22081665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DNMT3A R882</td>\n",
       "      <td>32</td>\n",
       "      <td>Acute Myeloid Leukemia</td>\n",
       "      <td>9119</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Young AML patients (&lt;60 years old) with DNMT3A...</td>\n",
       "      <td>22490330</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  molecular_profile  molecular_profile_id                         disease  \\\n",
       "0        JAK2 V617F                    64               Lymphoid Leukemia   \n",
       "1      PDGFRA D842V                    99  Gastrointestinal Stromal Tumor   \n",
       "2       DNMT3A R882                    32          Acute Myeloid Leukemia   \n",
       "3       DNMT3A R882                    32          Acute Myeloid Leukemia   \n",
       "\n",
       "   doid evidence_type                                 evidence_statement  \\\n",
       "0  1037    Diagnostic  JAK2 V617F is not associated with lymphoid leu...   \n",
       "1  9253    Diagnostic  GIST tumors harboring PDGFRA D842V mutation ar...   \n",
       "2  9119    Diagnostic  DNMT3A R882 mutations occur most often in de n...   \n",
       "3  9119    Diagnostic  Young AML patients (<60 years old) with DNMT3A...   \n",
       "\n",
       "  citation_id  \n",
       "0    16081687  \n",
       "1    15146165  \n",
       "2    22081665  \n",
       "3    22490330  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "raw_df = raw_df.drop(columns=[\"phenotypes\", \"therapies\", \"therapy_interaction_type\", \"evidence_direction\", \"evidence_level\", \n",
    "                              \"significance\", \"source_type\", 'asco_abstract_id', 'citation', 'nct_ids', 'rating', \n",
    "                              'evidence_status', 'evidence_id', 'variant_origin', 'last_review_date', 'evidence_civic_url', \n",
    "                              'molecular_profile_civic_url', 'is_flagged'])\n",
    "\n",
    "raw_df.head(4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "65951dd0-0bc3-4b35-814e-374b8db6f057",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(4282, 7)"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#drop any possible duplicates\n",
    "raw_df.drop_duplicates(subset=None, keep=\"first\", inplace=True)\n",
    "raw_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "f8ea3915-fa37-490c-845b-8296d8100617",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "                                      molecular_profile  molecular_profile_id  \\\n",
      "64                            BRAF V600E AND BRAF V600M                  4170   \n",
      "80                    BRAF V600E AND BRAF Amplification                  4173   \n",
      "124                        ALK EML4::ALK AND ALK L1196M                  4230   \n",
      "205                        ABL1 BCR::ABL AND ABL1 E255K                  4431   \n",
      "206                        ABL1 BCR::ABL AND ABL1 T315I                  4373   \n",
      "...                                                 ...                   ...   \n",
      "4271  CCND1 Amplification OR CCND2 Amplification OR ...                  4709   \n",
      "4274                             MSH2 Loss OR MLH1 Loss                  4733   \n",
      "4277  BRAF V600E OR NRAS Mutation OR HRAS Mutation O...                  4715   \n",
      "4286  BRAF V600E OR NRAS Mutation OR HRAS Mutation O...                  4748   \n",
      "4300                     ARID1A Loss OR ARID1A Wildtype                  4882   \n",
      "\n",
      "                            disease   doid evidence_type  \\\n",
      "64                         Melanoma   1909    Predictive   \n",
      "80                Colorectal Cancer   9256    Predictive   \n",
      "124   Lung Non-small Cell Carcinoma   3908    Predictive   \n",
      "205        Chronic Myeloid Leukemia   8552    Predictive   \n",
      "206        Chronic Myeloid Leukemia   8552    Predictive   \n",
      "...                             ...    ...           ...   \n",
      "4271                         Cancer    162    Predictive   \n",
      "4274                         Cancer    162    Predictive   \n",
      "4277                         Cancer    162    Predictive   \n",
      "4286                         Cancer    162    Predictive   \n",
      "4300   Ovarian Clear Cell Carcinoma  50934    Predictive   \n",
      "\n",
      "                                     evidence_statement citation_id  \n",
      "64    A single 66-year old male patient with advance...    23031422  \n",
      "80    COLO201 and COLO206F cells harboring BRAF V600...    21098728  \n",
      "124   CH5424802 treatment resulted in significant tu...    21575866  \n",
      "205   COS7 cell lines transfected with BCR-ABL const...    15194504  \n",
      "206   In chronic myeloid leukemia patients with the ...    20537386  \n",
      "...                                                 ...         ...  \n",
      "4271  This Phase 2 NCI-MATCH Trial subprotocol Z1B w...    36853016  \n",
      "4274  In this NCI-MATCH trial, 42 patients with 19 d...    31765263  \n",
      "4277  In this phase II study of the arm E of the NCI...    35363510  \n",
      "4286  In this Phase II study, 20 patients harboring ...    35363510  \n",
      "4300  In this phase 2 clinical study, 28 patients wi...    37418832  \n",
      "\n",
      "[362 rows x 7 columns]\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(362, 7)"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Make a dataframe that contains more that one biomarker per row\n",
    "AND_df = raw_df[raw_df[\"molecular_profile\"].str.contains(\" AND | OR \", case=False)]\n",
    "print(AND_df)\n",
    "AND_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "f82d039c-fe39-4a4f-b4b4-9dd93301372f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>molecular_profile</th>\n",
       "      <th>molecular_profile_id</th>\n",
       "      <th>disease</th>\n",
       "      <th>doid</th>\n",
       "      <th>evidence_type</th>\n",
       "      <th>evidence_statement</th>\n",
       "      <th>citation_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>BRAF V600E AND BRAF V600M</td>\n",
       "      <td>4170</td>\n",
       "      <td>Melanoma</td>\n",
       "      <td>1909</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>A single 66-year old male patient with advance...</td>\n",
       "      <td>23031422</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>BRAF V600E AND BRAF Amplification</td>\n",
       "      <td>4173</td>\n",
       "      <td>Colorectal Cancer</td>\n",
       "      <td>9256</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>COLO201 and COLO206F cells harboring BRAF V600...</td>\n",
       "      <td>21098728</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>124</th>\n",
       "      <td>ALK EML4::ALK AND ALK L1196M</td>\n",
       "      <td>4230</td>\n",
       "      <td>Lung Non-small Cell Carcinoma</td>\n",
       "      <td>3908</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>CH5424802 treatment resulted in significant tu...</td>\n",
       "      <td>21575866</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>205</th>\n",
       "      <td>ABL1 BCR::ABL AND ABL1 E255K</td>\n",
       "      <td>4431</td>\n",
       "      <td>Chronic Myeloid Leukemia</td>\n",
       "      <td>8552</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>COS7 cell lines transfected with BCR-ABL const...</td>\n",
       "      <td>15194504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>206</th>\n",
       "      <td>ABL1 BCR::ABL AND ABL1 T315I</td>\n",
       "      <td>4373</td>\n",
       "      <td>Chronic Myeloid Leukemia</td>\n",
       "      <td>8552</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>In chronic myeloid leukemia patients with the ...</td>\n",
       "      <td>20537386</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     molecular_profile  molecular_profile_id  \\\n",
       "64           BRAF V600E AND BRAF V600M                  4170   \n",
       "80   BRAF V600E AND BRAF Amplification                  4173   \n",
       "124       ALK EML4::ALK AND ALK L1196M                  4230   \n",
       "205       ABL1 BCR::ABL AND ABL1 E255K                  4431   \n",
       "206       ABL1 BCR::ABL AND ABL1 T315I                  4373   \n",
       "\n",
       "                           disease  doid evidence_type  \\\n",
       "64                        Melanoma  1909    Predictive   \n",
       "80               Colorectal Cancer  9256    Predictive   \n",
       "124  Lung Non-small Cell Carcinoma  3908    Predictive   \n",
       "205       Chronic Myeloid Leukemia  8552    Predictive   \n",
       "206       Chronic Myeloid Leukemia  8552    Predictive   \n",
       "\n",
       "                                    evidence_statement citation_id  \n",
       "64   A single 66-year old male patient with advance...    23031422  \n",
       "80   COLO201 and COLO206F cells harboring BRAF V600...    21098728  \n",
       "124  CH5424802 treatment resulted in significant tu...    21575866  \n",
       "205  COS7 cell lines transfected with BCR-ABL const...    15194504  \n",
       "206  In chronic myeloid leukemia patients with the ...    20537386  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Make a dataframe to separate the biomarkers\n",
    "AND1_df = AND_df.copy()\n",
    "AND2_df = AND_df.copy()\n",
    "AND3_df = AND_df.copy()\n",
    "AND4_df = AND_df.copy()\n",
    "AND5_df = AND_df.copy()\n",
    "AND6_df = AND_df.copy()\n",
    "\n",
    "AND3_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "d23ec6ee-4ff4-4c03-bc9c-53aa67bdb512",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(362, 7)"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Split biomarkers\n",
    "sep_df = AND_df[\"molecular_profile\"].str.split(\" AND | OR \", expand=True)\n",
    "sep_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "ff3dbf59-9137-4518-a4cc-941ab5705bb0",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\vania\\AppData\\Local\\Temp\\ipykernel_31392\\1309626977.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  AND_df[\"molecular_profile\"] = sep_df[0]\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "(362, 7)"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Put each biomarker in one dataframe\n",
    "AND_df[\"molecular_profile\"] = sep_df[0]\n",
    "AND_df.shape\n",
    "AND1_df[\"molecular_profile\"] = sep_df[1]\n",
    "AND1_df.shape\n",
    "AND2_df[\"molecular_profile\"] = sep_df[2]\n",
    "AND2_df.shape\n",
    "AND3_df[\"molecular_profile\"] = sep_df[3]\n",
    "AND3_df.shape\n",
    "AND4_df[\"molecular_profile\"] = sep_df[4]\n",
    "AND4_df.shape\n",
    "AND5_df[\"molecular_profile\"] = sep_df[5]\n",
    "AND5_df.shape\n",
    "AND6_df[\"molecular_profile\"] = sep_df[6]\n",
    "AND6_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "55524df8-5cf1-4526-9cf9-4c7fbf7e8e54",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>molecular_profile</th>\n",
       "      <th>molecular_profile_id</th>\n",
       "      <th>disease</th>\n",
       "      <th>doid</th>\n",
       "      <th>evidence_type</th>\n",
       "      <th>evidence_statement</th>\n",
       "      <th>citation_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BRAF V600E</td>\n",
       "      <td>4170</td>\n",
       "      <td>Melanoma</td>\n",
       "      <td>1909</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>A single 66-year old male patient with advance...</td>\n",
       "      <td>23031422</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BRAF V600E</td>\n",
       "      <td>4173</td>\n",
       "      <td>Colorectal Cancer</td>\n",
       "      <td>9256</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>COLO201 and COLO206F cells harboring BRAF V600...</td>\n",
       "      <td>21098728</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ALK EML4::ALK</td>\n",
       "      <td>4230</td>\n",
       "      <td>Lung Non-small Cell Carcinoma</td>\n",
       "      <td>3908</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>CH5424802 treatment resulted in significant tu...</td>\n",
       "      <td>21575866</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ABL1 BCR::ABL</td>\n",
       "      <td>4431</td>\n",
       "      <td>Chronic Myeloid Leukemia</td>\n",
       "      <td>8552</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>COS7 cell lines transfected with BCR-ABL const...</td>\n",
       "      <td>15194504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ABL1 BCR::ABL</td>\n",
       "      <td>4373</td>\n",
       "      <td>Chronic Myeloid Leukemia</td>\n",
       "      <td>8552</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>In chronic myeloid leukemia patients with the ...</td>\n",
       "      <td>20537386</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  molecular_profile  molecular_profile_id                        disease  \\\n",
       "0        BRAF V600E                  4170                       Melanoma   \n",
       "1        BRAF V600E                  4173              Colorectal Cancer   \n",
       "2     ALK EML4::ALK                  4230  Lung Non-small Cell Carcinoma   \n",
       "3     ABL1 BCR::ABL                  4431       Chronic Myeloid Leukemia   \n",
       "4     ABL1 BCR::ABL                  4373       Chronic Myeloid Leukemia   \n",
       "\n",
       "   doid evidence_type                                 evidence_statement  \\\n",
       "0  1909    Predictive  A single 66-year old male patient with advance...   \n",
       "1  9256    Predictive  COLO201 and COLO206F cells harboring BRAF V600...   \n",
       "2  3908    Predictive  CH5424802 treatment resulted in significant tu...   \n",
       "3  8552    Predictive  COS7 cell lines transfected with BCR-ABL const...   \n",
       "4  8552    Predictive  In chronic myeloid leukemia patients with the ...   \n",
       "\n",
       "  citation_id  \n",
       "0    23031422  \n",
       "1    21098728  \n",
       "2    21575866  \n",
       "3    15194504  \n",
       "4    20537386  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Join dataframes\n",
    "newsep_df = pd.concat([AND_df, AND1_df, AND2_df, AND3_df, AND4_df, AND5_df, AND6_df], ignore_index=True, sort=False)\n",
    "newsep_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "7139f154-180f-4f39-ad7f-2c60ac0bade2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(742, 7)"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Delete rows with no biomarker\n",
    "newsep_df = newsep_df.dropna(subset=[\"molecular_profile\"])\n",
    "newsep_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "4ae036b1-ff8f-4897-b7f4-90d5bdb85a3b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(3925, 7)"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Remove rows that contain more than one biomarker\n",
    "single_df = raw_df[~raw_df[\"molecular_profile\"].str.contains(\" AND | OR \")]\n",
    "single_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "7bf9c2b6-8b1f-4fb0-a11a-8bd129715a24",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(4667, 7)"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Add rows with the separated biomarkers\n",
    "single_df = pd.concat([single_df, newsep_df], ignore_index=True, sort=False)\n",
    "single_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "44e7083d-7cf1-4815-8e7b-440a10a950bf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(4662, 7)"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Eliminate duplicated rows\n",
    "single_df.drop_duplicates(subset=None, keep=\"first\", inplace=True)\n",
    "single_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "9b3e33ae-389f-499b-823e-171cf8f1e214",
   "metadata": {},
   "outputs": [],
   "source": [
    "# List rows with biomarkers joined by and/or and download for manual curation\n",
    "andor_df = single_df[single_df[\"molecular_profile\"].str.contains(\" and | or \")]\n",
    "andor_df.to_csv('andor.tsv', sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "abcefcc6-6f32-482e-ba08-2a099a4d4445",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>molecular_profile</th>\n",
       "      <th>molecular_profile_id</th>\n",
       "      <th>disease</th>\n",
       "      <th>doid</th>\n",
       "      <th>evidence_type</th>\n",
       "      <th>evidence_statement</th>\n",
       "      <th>citation_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NTRK1 LMNA::NTRK1 G595R</td>\n",
       "      <td>1256</td>\n",
       "      <td>Colorectal Adenocarcinoma</td>\n",
       "      <td>50861</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>A gene fusion between exon 10 of NTRK1 and exo...</td>\n",
       "      <td>26546295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NTRK1 LMNA::NTRK1 G667C</td>\n",
       "      <td>1256</td>\n",
       "      <td>Colorectal Adenocarcinoma</td>\n",
       "      <td>50861</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>A gene fusion between exon 10 of NTRK1 and exo...</td>\n",
       "      <td>26546295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PIK3CA H1047L</td>\n",
       "      <td>3634</td>\n",
       "      <td>Breast Tubular Carcinoma</td>\n",
       "      <td>6587</td>\n",
       "      <td>Prognostic</td>\n",
       "      <td>In a retrospective analysis of 32 Western Chin...</td>\n",
       "      <td>28269754</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PIK3CA H1047R</td>\n",
       "      <td>3634</td>\n",
       "      <td>Breast Tubular Carcinoma</td>\n",
       "      <td>6587</td>\n",
       "      <td>Prognostic</td>\n",
       "      <td>In a retrospective analysis of 32 Western Chin...</td>\n",
       "      <td>28269754</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PIK3CA Exon 10 mutation</td>\n",
       "      <td>1867</td>\n",
       "      <td>Colorectal Cancer</td>\n",
       "      <td>9256</td>\n",
       "      <td>Prognostic</td>\n",
       "      <td>This was a prospective study involving 1,170 c...</td>\n",
       "      <td>22357840</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         molecular_profile  molecular_profile_id                    disease  \\\n",
       "0  NTRK1 LMNA::NTRK1 G595R                  1256  Colorectal Adenocarcinoma   \n",
       "1  NTRK1 LMNA::NTRK1 G667C                  1256  Colorectal Adenocarcinoma   \n",
       "2            PIK3CA H1047L                  3634   Breast Tubular Carcinoma   \n",
       "3            PIK3CA H1047R                  3634   Breast Tubular Carcinoma   \n",
       "4  PIK3CA Exon 10 mutation                  1867          Colorectal Cancer   \n",
       "\n",
       "    doid evidence_type                                 evidence_statement  \\\n",
       "0  50861    Predictive  A gene fusion between exon 10 of NTRK1 and exo...   \n",
       "1  50861    Predictive  A gene fusion between exon 10 of NTRK1 and exo...   \n",
       "2   6587    Prognostic  In a retrospective analysis of 32 Western Chin...   \n",
       "3   6587    Prognostic  In a retrospective analysis of 32 Western Chin...   \n",
       "4   9256    Prognostic  This was a prospective study involving 1,170 c...   \n",
       "\n",
       "   citation_id  \n",
       "0     26546295  \n",
       "1     26546295  \n",
       "2     28269754  \n",
       "3     28269754  \n",
       "4     22357840  "
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Upload manually separated and/or biomarkers\n",
    "andor_curated = pd.read_csv('andor_curated.tsv', sep='\\t')\n",
    "andor_curated.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "2183d99b-af53-473c-b786-c74b4c6d391a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(4667, 7)"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Eliminate and/or rows from dataset and include newuly separated rows\n",
    "single_df = single_df[~single_df[\"molecular_profile\"].str.contains(\" and | or \")]\n",
    "single_df = pd.concat([single_df, andor_curated], ignore_index=True, sort=False)\n",
    "single_df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "f2b37af7-1234-42bb-acd3-e327d8209ace",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>molecular_profile</th>\n",
       "      <th>molecular_profile_id</th>\n",
       "      <th>disease</th>\n",
       "      <th>doid</th>\n",
       "      <th>evidence_type</th>\n",
       "      <th>evidence_statement</th>\n",
       "      <th>citation_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4657</th>\n",
       "      <td>NTRK1 LMNA::NTRK1 G595R</td>\n",
       "      <td>1256</td>\n",
       "      <td>Colorectal Adenocarcinoma</td>\n",
       "      <td>50861</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>A gene fusion between exon 10 of NTRK1 and exo...</td>\n",
       "      <td>26546295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4658</th>\n",
       "      <td>NTRK1 LMNA::NTRK1 G667C</td>\n",
       "      <td>1256</td>\n",
       "      <td>Colorectal Adenocarcinoma</td>\n",
       "      <td>50861</td>\n",
       "      <td>Predictive</td>\n",
       "      <td>A gene fusion between exon 10 of NTRK1 and exo...</td>\n",
       "      <td>26546295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4659</th>\n",
       "      <td>PIK3CA H1047L</td>\n",
       "      <td>3634</td>\n",
       "      <td>Breast Tubular Carcinoma</td>\n",
       "      <td>6587</td>\n",
       "      <td>Prognostic</td>\n",
       "      <td>In a retrospective analysis of 32 Western Chin...</td>\n",
       "      <td>28269754</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4660</th>\n",
       "      <td>PIK3CA H1047R</td>\n",
       "      <td>3634</td>\n",
       "      <td>Breast Tubular Carcinoma</td>\n",
       "      <td>6587</td>\n",
       "      <td>Prognostic</td>\n",
       "      <td>In a retrospective analysis of 32 Western Chin...</td>\n",
       "      <td>28269754</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4661</th>\n",
       "      <td>PIK3CA Exon 10 mutation</td>\n",
       "      <td>1867</td>\n",
       "      <td>Colorectal Cancer</td>\n",
       "      <td>9256</td>\n",
       "      <td>Prognostic</td>\n",
       "      <td>This was a prospective study involving 1,170 c...</td>\n",
       "      <td>22357840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4662</th>\n",
       "      <td>PIK3CA Exon 21 mutation</td>\n",
       "      <td>1867</td>\n",
       "      <td>Colorectal Cancer</td>\n",
       "      <td>9256</td>\n",
       "      <td>Prognostic</td>\n",
       "      <td>This was a prospective study involving 1,170 c...</td>\n",
       "      <td>22357840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4663</th>\n",
       "      <td>PIK3CA Exon 10 mutation</td>\n",
       "      <td>3179</td>\n",
       "      <td>Rectum Cancer</td>\n",
       "      <td>1993</td>\n",
       "      <td>Prognostic</td>\n",
       "      <td>This study examined outcomes of 240 rectum can...</td>\n",
       "      <td>19903786</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4664</th>\n",
       "      <td>PIK3CA Exon 21 mutation</td>\n",
       "      <td>3179</td>\n",
       "      <td>Rectum Cancer</td>\n",
       "      <td>1993</td>\n",
       "      <td>Prognostic</td>\n",
       "      <td>This study examined outcomes of 240 rectum can...</td>\n",
       "      <td>19903786</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4665</th>\n",
       "      <td>IKZF1 IKZF1 deletion</td>\n",
       "      <td>2613</td>\n",
       "      <td>Childhood B-cell Acute Lymphoblastic Leukemia</td>\n",
       "      <td>80146</td>\n",
       "      <td>Prognostic</td>\n",
       "      <td>This study reports on two pediatric cohorts wi...</td>\n",
       "      <td>19129520</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4666</th>\n",
       "      <td>IKZF1 IKZF1 mutation</td>\n",
       "      <td>2613</td>\n",
       "      <td>Childhood B-cell Acute Lymphoblastic Leukemia</td>\n",
       "      <td>80146</td>\n",
       "      <td>Prognostic</td>\n",
       "      <td>This study reports on two pediatric cohorts wi...</td>\n",
       "      <td>19129520</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            molecular_profile  molecular_profile_id  \\\n",
       "4657  NTRK1 LMNA::NTRK1 G595R                  1256   \n",
       "4658  NTRK1 LMNA::NTRK1 G667C                  1256   \n",
       "4659            PIK3CA H1047L                  3634   \n",
       "4660            PIK3CA H1047R                  3634   \n",
       "4661  PIK3CA Exon 10 mutation                  1867   \n",
       "4662  PIK3CA Exon 21 mutation                  1867   \n",
       "4663  PIK3CA Exon 10 mutation                  3179   \n",
       "4664  PIK3CA Exon 21 mutation                  3179   \n",
       "4665     IKZF1 IKZF1 deletion                  2613   \n",
       "4666     IKZF1 IKZF1 mutation                  2613   \n",
       "\n",
       "                                            disease   doid evidence_type  \\\n",
       "4657                      Colorectal Adenocarcinoma  50861    Predictive   \n",
       "4658                      Colorectal Adenocarcinoma  50861    Predictive   \n",
       "4659                       Breast Tubular Carcinoma   6587    Prognostic   \n",
       "4660                       Breast Tubular Carcinoma   6587    Prognostic   \n",
       "4661                              Colorectal Cancer   9256    Prognostic   \n",
       "4662                              Colorectal Cancer   9256    Prognostic   \n",
       "4663                                  Rectum Cancer   1993    Prognostic   \n",
       "4664                                  Rectum Cancer   1993    Prognostic   \n",
       "4665  Childhood B-cell Acute Lymphoblastic Leukemia  80146    Prognostic   \n",
       "4666  Childhood B-cell Acute Lymphoblastic Leukemia  80146    Prognostic   \n",
       "\n",
       "                                     evidence_statement citation_id  \n",
       "4657  A gene fusion between exon 10 of NTRK1 and exo...    26546295  \n",
       "4658  A gene fusion between exon 10 of NTRK1 and exo...    26546295  \n",
       "4659  In a retrospective analysis of 32 Western Chin...    28269754  \n",
       "4660  In a retrospective analysis of 32 Western Chin...    28269754  \n",
       "4661  This was a prospective study involving 1,170 c...    22357840  \n",
       "4662  This was a prospective study involving 1,170 c...    22357840  \n",
       "4663  This study examined outcomes of 240 rectum can...    19903786  \n",
       "4664  This study examined outcomes of 240 rectum can...    19903786  \n",
       "4665  This study reports on two pediatric cohorts wi...    19129520  \n",
       "4666  This study reports on two pediatric cohorts wi...    19129520  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "single_df.tail(10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "e13c2846-aae1-4163-90df-b8c448bf5fe2",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>molecular_profile</th>\n",
       "      <th>molecular_profile_id</th>\n",
       "      <th>disease</th>\n",
       "      <th>doid</th>\n",
       "      <th>evidence_type</th>\n",
       "      <th>evidence_statement</th>\n",
       "      <th>citation_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>JAK2 V617F mutation</td>\n",
       "      <td>64</td>\n",
       "      <td>Lymphoid Leukemia</td>\n",
       "      <td>1037</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>JAK2 V617F is not associated with lymphoid leu...</td>\n",
       "      <td>16081687</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PDGFRA D842V mutation</td>\n",
       "      <td>99</td>\n",
       "      <td>Gastrointestinal Stromal Tumor</td>\n",
       "      <td>9253</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>GIST tumors harboring PDGFRA D842V mutation ar...</td>\n",
       "      <td>15146165</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DNMT3A R882 mutation</td>\n",
       "      <td>32</td>\n",
       "      <td>Acute Myeloid Leukemia</td>\n",
       "      <td>9119</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>DNMT3A R882 mutations occur most often in de n...</td>\n",
       "      <td>22081665</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DNMT3A R882 mutation</td>\n",
       "      <td>32</td>\n",
       "      <td>Acute Myeloid Leukemia</td>\n",
       "      <td>9119</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Young AML patients (&lt;60 years old) with DNMT3A...</td>\n",
       "      <td>22490330</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JAK2 V617F mutation</td>\n",
       "      <td>64</td>\n",
       "      <td>Chronic Myeloid Leukemia</td>\n",
       "      <td>8552</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>JAK2 V617F is associated with myeloid malignan...</td>\n",
       "      <td>16081687</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       molecular_profile  molecular_profile_id  \\\n",
       "0    JAK2 V617F mutation                    64   \n",
       "1  PDGFRA D842V mutation                    99   \n",
       "2   DNMT3A R882 mutation                    32   \n",
       "3   DNMT3A R882 mutation                    32   \n",
       "4    JAK2 V617F mutation                    64   \n",
       "\n",
       "                          disease  doid evidence_type  \\\n",
       "0               Lymphoid Leukemia  1037    Diagnostic   \n",
       "1  Gastrointestinal Stromal Tumor  9253    Diagnostic   \n",
       "2          Acute Myeloid Leukemia  9119    Diagnostic   \n",
       "3          Acute Myeloid Leukemia  9119    Diagnostic   \n",
       "4        Chronic Myeloid Leukemia  8552    Diagnostic   \n",
       "\n",
       "                                  evidence_statement citation_id  \n",
       "0  JAK2 V617F is not associated with lymphoid leu...    16081687  \n",
       "1  GIST tumors harboring PDGFRA D842V mutation ar...    15146165  \n",
       "2  DNMT3A R882 mutations occur most often in de n...    22081665  \n",
       "3  Young AML patients (<60 years old) with DNMT3A...    22490330  \n",
       "4  JAK2 V617F is associated with myeloid malignan...    16081687  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Add the word mutation to rows with only gene and mutation symbols\n",
    "skip_list = [\"mutation\", \"overexpression\", \"expression\", \"amplification\", \"rearrangement\", \"loss\", \"underexpression\", \n",
    "             \"wildtype\", \"exon\", \"frameshift\", \"truncation\", \"promoter\", \"hypermethylation\", \"phosphorylation\", \"methylation\",\n",
    "             \"alteration\", \"fusion\", \"gain-of-function\", \"del\", \"ins\"]\n",
    "single_df['molecular_profile'] = single_df['molecular_profile'].apply(lambda x: x + ' mutation' \n",
    "                                          if len(x.split()) < 3 \n",
    "                                          and all(word not in x.lower() for word in skip_list)\n",
    "                                                  else x)\n",
    "single_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "a7524809-bbb7-4289-bf07-e22727123f71",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>molecular_profile</th>\n",
       "      <th>molecular_profile_id</th>\n",
       "      <th>disease</th>\n",
       "      <th>doid</th>\n",
       "      <th>evidence_type</th>\n",
       "      <th>evidence_statement</th>\n",
       "      <th>citation_id</th>\n",
       "      <th>gene</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>JAK2 V617F mutation</td>\n",
       "      <td>64</td>\n",
       "      <td>Lymphoid Leukemia</td>\n",
       "      <td>1037</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>JAK2 V617F is not associated with lymphoid leu...</td>\n",
       "      <td>16081687</td>\n",
       "      <td>JAK2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PDGFRA D842V mutation</td>\n",
       "      <td>99</td>\n",
       "      <td>Gastrointestinal Stromal Tumor</td>\n",
       "      <td>9253</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>GIST tumors harboring PDGFRA D842V mutation ar...</td>\n",
       "      <td>15146165</td>\n",
       "      <td>PDGFRA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DNMT3A R882 mutation</td>\n",
       "      <td>32</td>\n",
       "      <td>Acute Myeloid Leukemia</td>\n",
       "      <td>9119</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>DNMT3A R882 mutations occur most often in de n...</td>\n",
       "      <td>22081665</td>\n",
       "      <td>DNMT3A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DNMT3A R882 mutation</td>\n",
       "      <td>32</td>\n",
       "      <td>Acute Myeloid Leukemia</td>\n",
       "      <td>9119</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Young AML patients (&lt;60 years old) with DNMT3A...</td>\n",
       "      <td>22490330</td>\n",
       "      <td>DNMT3A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JAK2 V617F mutation</td>\n",
       "      <td>64</td>\n",
       "      <td>Chronic Myeloid Leukemia</td>\n",
       "      <td>8552</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>JAK2 V617F is associated with myeloid malignan...</td>\n",
       "      <td>16081687</td>\n",
       "      <td>JAK2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       molecular_profile  molecular_profile_id  \\\n",
       "0    JAK2 V617F mutation                    64   \n",
       "1  PDGFRA D842V mutation                    99   \n",
       "2   DNMT3A R882 mutation                    32   \n",
       "3   DNMT3A R882 mutation                    32   \n",
       "4    JAK2 V617F mutation                    64   \n",
       "\n",
       "                          disease  doid evidence_type  \\\n",
       "0               Lymphoid Leukemia  1037    Diagnostic   \n",
       "1  Gastrointestinal Stromal Tumor  9253    Diagnostic   \n",
       "2          Acute Myeloid Leukemia  9119    Diagnostic   \n",
       "3          Acute Myeloid Leukemia  9119    Diagnostic   \n",
       "4        Chronic Myeloid Leukemia  8552    Diagnostic   \n",
       "\n",
       "                                  evidence_statement citation_id    gene  \n",
       "0  JAK2 V617F is not associated with lymphoid leu...    16081687    JAK2  \n",
       "1  GIST tumors harboring PDGFRA D842V mutation ar...    15146165  PDGFRA  \n",
       "2  DNMT3A R882 mutations occur most often in de n...    22081665  DNMT3A  \n",
       "3  Young AML patients (<60 years old) with DNMT3A...    22490330  DNMT3A  \n",
       "4  JAK2 V617F is associated with myeloid malignan...    16081687    JAK2  "
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#extracting the gene symbol\n",
    "gene_split = single_df[\"molecular_profile\"].str.split(\" \",n=1, expand=True)\n",
    "single_df[\"gene\"] = gene_split[0]\n",
    "single_df.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "7b524922-06e3-420c-a7e2-41fa42fb3f66",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Download gene col and convert the gene symbol list to gene id with NCBI DAVID conversion tool: https://david.ncifcrf.gov/conversion.jsp\n",
    "single_df[\"gene\"].to_csv(\"gene_list.tsv\", sep=\"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "51625d3f-56c1-4bbd-8cb6-f0863b03d046",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    From     To       Species  \\\n",
      "0   ARAF    369  Homo sapiens   \n",
      "1   CBLC  23624  Homo sapiens   \n",
      "2  IKZF1  10320  Homo sapiens   \n",
      "3  ALCAM    214  Homo sapiens   \n",
      "\n",
      "                                           Gene Name  \n",
      "0  A-Raf proto-oncogene, serine/threonine kinase(...  \n",
      "1                         Cbl proto-oncogene C(CBLC)  \n",
      "2                 IKAROS family zinc finger 1(IKZF1)  \n",
      "3  activated leukocyte cell adhesion molecule(ALCAM)  \n"
     ]
    }
   ],
   "source": [
    "#upload converted list\n",
    "david_geneid = pd.read_csv(\"david_geneid.txt\", sep='\\t')\n",
    "print(david_geneid.head(4))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "3e8b7515-70b7-497b-81b3-2e942c63b8f1",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>molecular_profile</th>\n",
       "      <th>molecular_profile_id</th>\n",
       "      <th>disease</th>\n",
       "      <th>doid</th>\n",
       "      <th>evidence_type</th>\n",
       "      <th>evidence_statement</th>\n",
       "      <th>citation_id</th>\n",
       "      <th>gene</th>\n",
       "      <th>To</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>JAK2 V617F mutation</td>\n",
       "      <td>64</td>\n",
       "      <td>Lymphoid Leukemia</td>\n",
       "      <td>1037</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>JAK2 V617F is not associated with lymphoid leu...</td>\n",
       "      <td>16081687</td>\n",
       "      <td>JAK2</td>\n",
       "      <td>3717</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PDGFRA D842V mutation</td>\n",
       "      <td>99</td>\n",
       "      <td>Gastrointestinal Stromal Tumor</td>\n",
       "      <td>9253</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>GIST tumors harboring PDGFRA D842V mutation ar...</td>\n",
       "      <td>15146165</td>\n",
       "      <td>PDGFRA</td>\n",
       "      <td>5156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DNMT3A R882 mutation</td>\n",
       "      <td>32</td>\n",
       "      <td>Acute Myeloid Leukemia</td>\n",
       "      <td>9119</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>DNMT3A R882 mutations occur most often in de n...</td>\n",
       "      <td>22081665</td>\n",
       "      <td>DNMT3A</td>\n",
       "      <td>1788</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DNMT3A R882 mutation</td>\n",
       "      <td>32</td>\n",
       "      <td>Acute Myeloid Leukemia</td>\n",
       "      <td>9119</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>Young AML patients (&lt;60 years old) with DNMT3A...</td>\n",
       "      <td>22490330</td>\n",
       "      <td>DNMT3A</td>\n",
       "      <td>1788</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       molecular_profile  molecular_profile_id  \\\n",
       "0    JAK2 V617F mutation                    64   \n",
       "1  PDGFRA D842V mutation                    99   \n",
       "2   DNMT3A R882 mutation                    32   \n",
       "3   DNMT3A R882 mutation                    32   \n",
       "\n",
       "                          disease  doid evidence_type  \\\n",
       "0               Lymphoid Leukemia  1037    Diagnostic   \n",
       "1  Gastrointestinal Stromal Tumor  9253    Diagnostic   \n",
       "2          Acute Myeloid Leukemia  9119    Diagnostic   \n",
       "3          Acute Myeloid Leukemia  9119    Diagnostic   \n",
       "\n",
       "                                  evidence_statement citation_id    gene    To  \n",
       "0  JAK2 V617F is not associated with lymphoid leu...    16081687    JAK2  3717  \n",
       "1  GIST tumors harboring PDGFRA D842V mutation ar...    15146165  PDGFRA  5156  \n",
       "2  DNMT3A R882 mutations occur most often in de n...    22081665  DNMT3A  1788  \n",
       "3  Young AML patients (<60 years old) with DNMT3A...    22490330  DNMT3A  1788  "
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Merging gene id data\n",
    "david_geneid = david_geneid.drop(columns=[\"Species\", \"Gene Name\"])\n",
    "david_geneid.rename(columns={\"From\":\"gene\"}, inplace=True)\n",
    "\n",
    "df_merged= pd.merge(single_df, david_geneid, on=\"gene\", how=\"left\")\n",
    "df_merged['To'] = df_merged['To'].fillna(0).astype('Int64')\n",
    "df_merged.head(4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "42de8d31-aa80-4e71-b88c-0e62d3eff8cf",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Biomarker</th>\n",
       "      <th>assessesed_biomarker_entity</th>\n",
       "      <th>assessed_biomarker_entity_ID</th>\n",
       "      <th>assessed_entity_type</th>\n",
       "      <th>condition</th>\n",
       "      <th>condition_ID</th>\n",
       "      <th>exposure_agent</th>\n",
       "      <th>exposure_agent_ID</th>\n",
       "      <th>best_biomarker_role</th>\n",
       "      <th>specimen</th>\n",
       "      <th>specimen_ID</th>\n",
       "      <th>loinc_code</th>\n",
       "      <th>evidence_source</th>\n",
       "      <th>evidence</th>\n",
       "      <th>evidence_source2</th>\n",
       "      <th>evidence2</th>\n",
       "      <th>tag</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>JAK2 V617F mutation</td>\n",
       "      <td>JAK2</td>\n",
       "      <td>3717</td>\n",
       "      <td>gene</td>\n",
       "      <td>Lymphoid Leukemia</td>\n",
       "      <td>1037</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>16081687</td>\n",
       "      <td>JAK2 V617F is not associated with lymphoid leu...</td>\n",
       "      <td>64</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PDGFRA D842V mutation</td>\n",
       "      <td>PDGFRA</td>\n",
       "      <td>5156</td>\n",
       "      <td>gene</td>\n",
       "      <td>Gastrointestinal Stromal Tumor</td>\n",
       "      <td>9253</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15146165</td>\n",
       "      <td>GIST tumors harboring PDGFRA D842V mutation ar...</td>\n",
       "      <td>99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DNMT3A R882 mutation</td>\n",
       "      <td>DNMT3A</td>\n",
       "      <td>1788</td>\n",
       "      <td>gene</td>\n",
       "      <td>Acute Myeloid Leukemia</td>\n",
       "      <td>9119</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22081665</td>\n",
       "      <td>DNMT3A R882 mutations occur most often in de n...</td>\n",
       "      <td>32</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DNMT3A R882 mutation</td>\n",
       "      <td>DNMT3A</td>\n",
       "      <td>1788</td>\n",
       "      <td>gene</td>\n",
       "      <td>Acute Myeloid Leukemia</td>\n",
       "      <td>9119</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22490330</td>\n",
       "      <td>Young AML patients (&lt;60 years old) with DNMT3A...</td>\n",
       "      <td>32</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JAK2 V617F mutation</td>\n",
       "      <td>JAK2</td>\n",
       "      <td>3717</td>\n",
       "      <td>gene</td>\n",
       "      <td>Chronic Myeloid Leukemia</td>\n",
       "      <td>8552</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>16081687</td>\n",
       "      <td>JAK2 V617F is associated with myeloid malignan...</td>\n",
       "      <td>64</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               Biomarker assessesed_biomarker_entity  \\\n",
       "0    JAK2 V617F mutation                        JAK2   \n",
       "1  PDGFRA D842V mutation                      PDGFRA   \n",
       "2   DNMT3A R882 mutation                      DNMT3A   \n",
       "3   DNMT3A R882 mutation                      DNMT3A   \n",
       "4    JAK2 V617F mutation                        JAK2   \n",
       "\n",
       "   assessed_biomarker_entity_ID assessed_entity_type  \\\n",
       "0                          3717                 gene   \n",
       "1                          5156                 gene   \n",
       "2                          1788                 gene   \n",
       "3                          1788                 gene   \n",
       "4                          3717                 gene   \n",
       "\n",
       "                        condition  condition_ID  exposure_agent  \\\n",
       "0               Lymphoid Leukemia          1037             NaN   \n",
       "1  Gastrointestinal Stromal Tumor          9253             NaN   \n",
       "2          Acute Myeloid Leukemia          9119             NaN   \n",
       "3          Acute Myeloid Leukemia          9119             NaN   \n",
       "4        Chronic Myeloid Leukemia          8552             NaN   \n",
       "\n",
       "   exposure_agent_ID best_biomarker_role  specimen  specimen_ID  loinc_code  \\\n",
       "0                NaN          Diagnostic       NaN          NaN         NaN   \n",
       "1                NaN          Diagnostic       NaN          NaN         NaN   \n",
       "2                NaN          Diagnostic       NaN          NaN         NaN   \n",
       "3                NaN          Diagnostic       NaN          NaN         NaN   \n",
       "4                NaN          Diagnostic       NaN          NaN         NaN   \n",
       "\n",
       "  evidence_source                                           evidence  \\\n",
       "0        16081687  JAK2 V617F is not associated with lymphoid leu...   \n",
       "1        15146165  GIST tumors harboring PDGFRA D842V mutation ar...   \n",
       "2        22081665  DNMT3A R882 mutations occur most often in de n...   \n",
       "3        22490330  Young AML patients (<60 years old) with DNMT3A...   \n",
       "4        16081687  JAK2 V617F is associated with myeloid malignan...   \n",
       "\n",
       "   evidence_source2  evidence2  \\\n",
       "0                64        NaN   \n",
       "1                99        NaN   \n",
       "2                32        NaN   \n",
       "3                32        NaN   \n",
       "4                64        NaN   \n",
       "\n",
       "                                                 tag  \n",
       "0  biomarker;assessed_biomarker_entity_id;assesse...  \n",
       "1  biomarker;assessed_biomarker_entity_id;assesse...  \n",
       "2  biomarker;assessed_biomarker_entity_id;assesse...  \n",
       "3  biomarker;assessed_biomarker_entity_id;assesse...  \n",
       "4  biomarker;assessed_biomarker_entity_id;assesse...  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Renaming and organizing columns to biomarker data model format\n",
    "df_merged = df_merged.reindex(columns=df_merged.columns.tolist() + ['exposure_agent', 'exposure_agent_ID', 'specimen', 'specimen_ID', 'loinc_code', \"evidence2\"])\n",
    "df_merged2 = df_merged[['molecular_profile', 'gene', 'To', 'disease', 'doid', 'exposure_agent', 'exposure_agent_ID',\n",
    "       'evidence_type','specimen', 'specimen_ID', 'loinc_code', 'citation_id','evidence_statement', \"molecular_profile_id\", \"evidence2\"]]\n",
    "df_merged2.columns = ['Biomarker', 'assessesed_biomarker_entity', 'assessed_biomarker_entity_ID','condition', 'condition_ID',\n",
    "                     'exposure_agent', 'exposure_agent_ID', 'best_biomarker_role', 'specimen', 'specimen_ID', 'loinc_code', 'evidence_source','evidence', 'evidence_source2','evidence2']\n",
    "df_merged2.insert(3, 'assessed_entity_type', \"gene\")\n",
    "df_merged2.insert(16, 'tag', 'biomarker;assessed_biomarker_entity_id;assessed_biomarker_entity;condition')\n",
    "\n",
    "df_merged2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "afa0439c-0c92-4d73-8b4f-513fa32c0a15",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\vania\\AppData\\Local\\Temp\\ipykernel_31392\\663848817.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_merged2[\"assessed_biomarker_entity_ID\"] = \"NCBI:\" + df_merged2[\"assessed_biomarker_entity_ID\"].astype(str)\n",
      "C:\\Users\\vania\\AppData\\Local\\Temp\\ipykernel_31392\\663848817.py:2: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_merged2[\"condition_ID\"] = \"DOID:\" + df_merged2[\"condition_ID\"].astype(str)\n",
      "C:\\Users\\vania\\AppData\\Local\\Temp\\ipykernel_31392\\663848817.py:3: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_merged2[\"evidence_source\"] = \"Pubmed:\" + df_merged2[\"evidence_source\"].astype(str)\n",
      "C:\\Users\\vania\\AppData\\Local\\Temp\\ipykernel_31392\\663848817.py:4: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df_merged2[\"evidence_source2\"] = \"CIVIC:\" + df_merged2[\"evidence_source2\"].astype(str)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Biomarker</th>\n",
       "      <th>assessesed_biomarker_entity</th>\n",
       "      <th>assessed_biomarker_entity_ID</th>\n",
       "      <th>assessed_entity_type</th>\n",
       "      <th>condition</th>\n",
       "      <th>condition_ID</th>\n",
       "      <th>exposure_agent</th>\n",
       "      <th>exposure_agent_ID</th>\n",
       "      <th>best_biomarker_role</th>\n",
       "      <th>specimen</th>\n",
       "      <th>specimen_ID</th>\n",
       "      <th>loinc_code</th>\n",
       "      <th>evidence_source</th>\n",
       "      <th>evidence</th>\n",
       "      <th>evidence_source2</th>\n",
       "      <th>evidence2</th>\n",
       "      <th>tag</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>JAK2 V617F mutation</td>\n",
       "      <td>JAK2</td>\n",
       "      <td>NCBI:3717</td>\n",
       "      <td>gene</td>\n",
       "      <td>Lymphoid Leukemia</td>\n",
       "      <td>DOID:1037</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pubmed:16081687</td>\n",
       "      <td>JAK2 V617F is not associated with lymphoid leu...</td>\n",
       "      <td>CIVIC:64</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PDGFRA D842V mutation</td>\n",
       "      <td>PDGFRA</td>\n",
       "      <td>NCBI:5156</td>\n",
       "      <td>gene</td>\n",
       "      <td>Gastrointestinal Stromal Tumor</td>\n",
       "      <td>DOID:9253</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pubmed:15146165</td>\n",
       "      <td>GIST tumors harboring PDGFRA D842V mutation ar...</td>\n",
       "      <td>CIVIC:99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DNMT3A R882 mutation</td>\n",
       "      <td>DNMT3A</td>\n",
       "      <td>NCBI:1788</td>\n",
       "      <td>gene</td>\n",
       "      <td>Acute Myeloid Leukemia</td>\n",
       "      <td>DOID:9119</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pubmed:22081665</td>\n",
       "      <td>DNMT3A R882 mutations occur most often in de n...</td>\n",
       "      <td>CIVIC:32</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DNMT3A R882 mutation</td>\n",
       "      <td>DNMT3A</td>\n",
       "      <td>NCBI:1788</td>\n",
       "      <td>gene</td>\n",
       "      <td>Acute Myeloid Leukemia</td>\n",
       "      <td>DOID:9119</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pubmed:22490330</td>\n",
       "      <td>Young AML patients (&lt;60 years old) with DNMT3A...</td>\n",
       "      <td>CIVIC:32</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JAK2 V617F mutation</td>\n",
       "      <td>JAK2</td>\n",
       "      <td>NCBI:3717</td>\n",
       "      <td>gene</td>\n",
       "      <td>Chronic Myeloid Leukemia</td>\n",
       "      <td>DOID:8552</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Pubmed:16081687</td>\n",
       "      <td>JAK2 V617F is associated with myeloid malignan...</td>\n",
       "      <td>CIVIC:64</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               Biomarker assessesed_biomarker_entity  \\\n",
       "0    JAK2 V617F mutation                        JAK2   \n",
       "1  PDGFRA D842V mutation                      PDGFRA   \n",
       "2   DNMT3A R882 mutation                      DNMT3A   \n",
       "3   DNMT3A R882 mutation                      DNMT3A   \n",
       "4    JAK2 V617F mutation                        JAK2   \n",
       "\n",
       "  assessed_biomarker_entity_ID assessed_entity_type  \\\n",
       "0                    NCBI:3717                 gene   \n",
       "1                    NCBI:5156                 gene   \n",
       "2                    NCBI:1788                 gene   \n",
       "3                    NCBI:1788                 gene   \n",
       "4                    NCBI:3717                 gene   \n",
       "\n",
       "                        condition condition_ID  exposure_agent  \\\n",
       "0               Lymphoid Leukemia    DOID:1037             NaN   \n",
       "1  Gastrointestinal Stromal Tumor    DOID:9253             NaN   \n",
       "2          Acute Myeloid Leukemia    DOID:9119             NaN   \n",
       "3          Acute Myeloid Leukemia    DOID:9119             NaN   \n",
       "4        Chronic Myeloid Leukemia    DOID:8552             NaN   \n",
       "\n",
       "   exposure_agent_ID best_biomarker_role  specimen  specimen_ID  loinc_code  \\\n",
       "0                NaN          Diagnostic       NaN          NaN         NaN   \n",
       "1                NaN          Diagnostic       NaN          NaN         NaN   \n",
       "2                NaN          Diagnostic       NaN          NaN         NaN   \n",
       "3                NaN          Diagnostic       NaN          NaN         NaN   \n",
       "4                NaN          Diagnostic       NaN          NaN         NaN   \n",
       "\n",
       "   evidence_source                                           evidence  \\\n",
       "0  Pubmed:16081687  JAK2 V617F is not associated with lymphoid leu...   \n",
       "1  Pubmed:15146165  GIST tumors harboring PDGFRA D842V mutation ar...   \n",
       "2  Pubmed:22081665  DNMT3A R882 mutations occur most often in de n...   \n",
       "3  Pubmed:22490330  Young AML patients (<60 years old) with DNMT3A...   \n",
       "4  Pubmed:16081687  JAK2 V617F is associated with myeloid malignan...   \n",
       "\n",
       "  evidence_source2  evidence2  \\\n",
       "0         CIVIC:64        NaN   \n",
       "1         CIVIC:99        NaN   \n",
       "2         CIVIC:32        NaN   \n",
       "3         CIVIC:32        NaN   \n",
       "4         CIVIC:64        NaN   \n",
       "\n",
       "                                                 tag  \n",
       "0  biomarker;assessed_biomarker_entity_id;assesse...  \n",
       "1  biomarker;assessed_biomarker_entity_id;assesse...  \n",
       "2  biomarker;assessed_biomarker_entity_id;assesse...  \n",
       "3  biomarker;assessed_biomarker_entity_id;assesse...  \n",
       "4  biomarker;assessed_biomarker_entity_id;assesse...  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_merged2[\"assessed_biomarker_entity_ID\"] = \"NCBI:\" + df_merged2[\"assessed_biomarker_entity_ID\"].astype(str)\n",
    "df_merged2[\"condition_ID\"] = \"DOID:\" + df_merged2[\"condition_ID\"].astype(str)\n",
    "df_merged2[\"evidence_source\"] = \"Pubmed:\" + df_merged2[\"evidence_source\"].astype(str)\n",
    "df_merged2[\"evidence_source2\"] = \"CIVIC:\" + df_merged2[\"evidence_source2\"].astype(str)\n",
    "\n",
    "df_merged2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "79d0b406-4910-41b9-ac08-cab00a0926e0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Biomarker</th>\n",
       "      <th>assessesed_biomarker_entity</th>\n",
       "      <th>assessed_biomarker_entity_ID</th>\n",
       "      <th>assessed_entity_type</th>\n",
       "      <th>condition</th>\n",
       "      <th>condition_ID</th>\n",
       "      <th>exposure_agent</th>\n",
       "      <th>exposure_agent_ID</th>\n",
       "      <th>best_biomarker_role</th>\n",
       "      <th>specimen</th>\n",
       "      <th>specimen_ID</th>\n",
       "      <th>loinc_code</th>\n",
       "      <th>evidence_source</th>\n",
       "      <th>evidence</th>\n",
       "      <th>tag</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>JAK2 V617F mutation</td>\n",
       "      <td>JAK2</td>\n",
       "      <td>NCBI:3717</td>\n",
       "      <td>gene</td>\n",
       "      <td>Lymphoid Leukemia</td>\n",
       "      <td>DOID:1037</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CIVIC:64</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PDGFRA D842V mutation</td>\n",
       "      <td>PDGFRA</td>\n",
       "      <td>NCBI:5156</td>\n",
       "      <td>gene</td>\n",
       "      <td>Gastrointestinal Stromal Tumor</td>\n",
       "      <td>DOID:9253</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CIVIC:99</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DNMT3A R882 mutation</td>\n",
       "      <td>DNMT3A</td>\n",
       "      <td>NCBI:1788</td>\n",
       "      <td>gene</td>\n",
       "      <td>Acute Myeloid Leukemia</td>\n",
       "      <td>DOID:9119</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CIVIC:32</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DNMT3A R882 mutation</td>\n",
       "      <td>DNMT3A</td>\n",
       "      <td>NCBI:1788</td>\n",
       "      <td>gene</td>\n",
       "      <td>Acute Myeloid Leukemia</td>\n",
       "      <td>DOID:9119</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CIVIC:32</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>JAK2 V617F mutation</td>\n",
       "      <td>JAK2</td>\n",
       "      <td>NCBI:3717</td>\n",
       "      <td>gene</td>\n",
       "      <td>Chronic Myeloid Leukemia</td>\n",
       "      <td>DOID:8552</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Diagnostic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CIVIC:64</td>\n",
       "      <td>NaN</td>\n",
       "      <td>biomarker;assessed_biomarker_entity_id;assesse...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               Biomarker assessesed_biomarker_entity  \\\n",
       "0    JAK2 V617F mutation                        JAK2   \n",
       "1  PDGFRA D842V mutation                      PDGFRA   \n",
       "2   DNMT3A R882 mutation                      DNMT3A   \n",
       "3   DNMT3A R882 mutation                      DNMT3A   \n",
       "4    JAK2 V617F mutation                        JAK2   \n",
       "\n",
       "  assessed_biomarker_entity_ID assessed_entity_type  \\\n",
       "0                    NCBI:3717                 gene   \n",
       "1                    NCBI:5156                 gene   \n",
       "2                    NCBI:1788                 gene   \n",
       "3                    NCBI:1788                 gene   \n",
       "4                    NCBI:3717                 gene   \n",
       "\n",
       "                        condition condition_ID  exposure_agent  \\\n",
       "0               Lymphoid Leukemia    DOID:1037             NaN   \n",
       "1  Gastrointestinal Stromal Tumor    DOID:9253             NaN   \n",
       "2          Acute Myeloid Leukemia    DOID:9119             NaN   \n",
       "3          Acute Myeloid Leukemia    DOID:9119             NaN   \n",
       "4        Chronic Myeloid Leukemia    DOID:8552             NaN   \n",
       "\n",
       "   exposure_agent_ID best_biomarker_role  specimen  specimen_ID  loinc_code  \\\n",
       "0                NaN          Diagnostic       NaN          NaN         NaN   \n",
       "1                NaN          Diagnostic       NaN          NaN         NaN   \n",
       "2                NaN          Diagnostic       NaN          NaN         NaN   \n",
       "3                NaN          Diagnostic       NaN          NaN         NaN   \n",
       "4                NaN          Diagnostic       NaN          NaN         NaN   \n",
       "\n",
       "  evidence_source  evidence                                                tag  \n",
       "0        CIVIC:64       NaN  biomarker;assessed_biomarker_entity_id;assesse...  \n",
       "1        CIVIC:99       NaN  biomarker;assessed_biomarker_entity_id;assesse...  \n",
       "2        CIVIC:32       NaN  biomarker;assessed_biomarker_entity_id;assesse...  \n",
       "3        CIVIC:32       NaN  biomarker;assessed_biomarker_entity_id;assesse...  \n",
       "4        CIVIC:64       NaN  biomarker;assessed_biomarker_entity_id;assesse...  "
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Duplicate the dataset and add one type of evidence to each duplicate\n",
    "pubmed_df1 = df_merged2.drop(columns=[\"evidence_source2\", \"evidence2\"])\n",
    "civic_df2 = df_merged2.drop(columns=[\"evidence_source\", \"evidence\"])\n",
    "civic_df2 = civic_df2.rename(columns = {\"evidence_source2\":\"evidence_source\", \"evidence2\":\"evidence\"})\n",
    "\n",
    "civic_df2.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "0c790008-685a-4b4d-9c1b-9a67f16857ae",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(9334, 15)"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Join civic and pubmed dfs\n",
    "civic_data = pd.concat([pubmed_df1, civic_df2], ignore_index=True, sort=False)\n",
    "\n",
    "civic_data.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "c96ef0f6-6bfc-4c82-a940-8d05d158dba7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(7396, 15)"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Drop duplicates\n",
    "civic_data = civic_data.drop_duplicates()\n",
    "civic_data.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "1883cfad-535c-4a98-b917-48a337857ee1",
   "metadata": {},
   "outputs": [],
   "source": [
    "civic_data.to_csv(\"civic_data.tsv\", sep= \"\\t\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3c424626-0b6f-49aa-8e48-c49550a1d33c",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
